batcher.go

```diff
@@ -0,0 +1,233 @@
+package batcher
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+)
+
+const (
+	DefaultMaxBatchSize = 100
+	DefaultMinBatchSize = 1
+	DefaultMaxHandlers  = 1
+	DefaultTicker       = 17 * time.Millisecond
+)
+
+type Result[V any] struct {
+	Value V
+	Err   error
+}
+
+func New[K comparable, V any](handler func(ctx context.Context, keys []K) []*Result[V], opts ...Option[K, V]) *batcher[K, V] {
```

Comment:
Exported function should return exported return type

Reply:
Are you talking about the public interface? I don't want to make the `batcher` public, because there is a lot of initialization in the new one.

Reply:
Fixed

Reply:
> It could be an interface to expose then

I liked the option when we restore the structure.
There is redundancy in this approach, you need to follow the initialization. But it is much more convenient to work with the structure